EV3D4WebControl in Mobile interface: AssertionError: None1050 is an invalid speed_pct, must be between -100 and 100 (inclusive)

I use ev3dev-stretch-ev3-generic-2018-08-06. Successfully downloaded the repo and tested out EV3D4WebControl in Desktop mode. However, when testing Mobile mode, the following error and its traceback is thrown:

Exception happened during processing of request from ('192.168.1.66', 60069)
Traceback (most recent call last):
  File "/usr/lib/python3.5/socketserver.py", line 313, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 341, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 354, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.5/socketserver.py", line 681, in __init__
    self.handle()
  File "/usr/lib/python3.5/http/server.py", line 422, in handle
    self.handle_one_request()
  File "/usr/lib/python3.5/http/server.py", line 410, in handle_one_request
    method()
  File "/usr/lib/python3/dist-packages/ev3dev2/control/webserver.py", line 212, in do_GET
    self.robot.on(x, y, motor_max_speed)
  File "/usr/lib/python3/dist-packages/ev3dev2/motor.py", line 2015, in on
    MoveTank.on(self, SpeedPercent(left_speed_percentage * self.left_motor._speed_pct(max_speed) / 100), SpeedPercent(right_speed_percentage * self.right_motor._speed_pct(max_speed) / 100))
  File "/usr/lib/python3/dist-packages/ev3dev2/motor.py", line 852, in _speed_pct
    "{}{} is an invalid speed_pct, must be between -100 and 100 (inclusive)".format(None if label is None else (label + ": ") , speed_pct)
AssertionError: None1050 is an invalid speed_pct, must be between -100 and 100 (inclusive)

--- a/debian/changelog
+++ b/debian/changelog
@@ -1,5 +1,6 @@
 python-ev3dev2 (2.0.0~beta3) stable; urgency=medium
 
+  * MoveJoyStick should use SpeedPercent
   * renamed GyroSensor rate_and_angle to angle_and_rate
 
  -- Kaelin Laundry <wasabifan@outlook.com>  Sat, 27 Oct 2018 21:18:00 -0700

--- a/ev3dev2/control/webserver.py
+++ b/ev3dev2/control/webserver.py
@@ -209,7 +209,7 @@ class TankWebHandler(RobotWebHandler):
 
             if joystick_engaged:
                 if seq > max_move_xy_seq:
-                    self.robot.on(x, y, motor_max_speed)
+                    self.robot.on(x, y)
                     max_move_xy_seq = seq
                     log.debug("seq %d: (x, y) (%4d, %4d)" % (seq, x, y))
                 else:

--- a/ev3dev2/motor.py
+++ b/ev3dev2/motor.py
@@ -1929,7 +1929,7 @@ class MoveJoystick(MoveTank):
     Used to control a pair of motors via a single joystick vector.
     """
 
-    def on(self, x, y, max_speed=100.0, radius=100.0):
+    def on(self, x, y, radius=100.0):
         """
         Convert x,y joystick coordinates to left/right motor speed percentages
         and move the motors.
@@ -1944,13 +1944,9 @@ class MoveJoystick(MoveTank):
             The X and Y coordinates of the joystick's position, with
             (0,0) representing the center position. X is horizontal and Y is vertical.
         
-        max_speed (default 100%):
-            A percentage or other SpeedValue, controlling the maximum motor speed.
-        
         radius (default 100):
             The radius of the joystick, controlling the range of the input (x, y) values.
             e.g. if "x" and "y" can be between -1 and 1, radius should be set to "1".
-        
         """
 
         # If joystick is in the middle stop the tank

--- a/ev3dev2/motor.py
+++ b/ev3dev2/motor.py
@@ -1987,8 +1987,9 @@ class MoveJoystick(MoveTank):
     #         init_left_speed_percentage, init_right_speed_percentage,
     #         left_speed_percentage, right_speed_percentage))
 
-        MoveTank.on(self, SpeedNativeUnits(left_speed_percentage / 100 * self.left_motor._speed_native_units(max_speed)), SpeedNativeUnits(right_speed_percentage / 100 * self.right_motor._speed_native_units(max_speed)))
-
+        MoveTank.on(self,
+            SpeedPercent(left_speed_percentage),
+            SpeedPercent(right_speed_percentage))
 
     @staticmethod
     def angle_to_speed_percentage(angle):

--- a/tests/api_tests.py
+++ b/tests/api_tests.py
@@ -316,10 +316,12 @@ class TestAPI(unittest.TestCase):
         populate_arena([('large_motor', 0, 'outA'), ('large_motor', 1, 'outB')])
 
         drive = MoveJoystick(OUTPUT_A, OUTPUT_B)
-        drive.on(100, 100, max_speed=SpeedPercent(50))
 
+        # With the joystick at (x, y) of (0, 50) we should drive straigh ahead
+        # at 50% of max_speed
+        drive.on(0, 50)
         self.assertEqual(drive.left_motor.speed_sp, 1050 / 2)
-        self.assertAlmostEqual(drive.right_motor.speed_sp, 0)
+        self.assertEqual(drive.right_motor.speed_sp, 1050 / 2)
 
     def test_units(self):
         clean_arena()

